projects/canopy/src/lib/filter-buttons/filter-group.component.ts

```diff
@@ -0,0 +1,176 @@
+  templateUrl: './filter-group.component.html',
+  styleUrls: ['./filter-group.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class LgFilterGroupComponent implements ControlValueAccessor {
+  @HostBinding('class.lg-filter-group--error') get errorClass() {
+    if (this.variant === 'select-one') {
+      return this.errorState.isControlInvalid(this.control, this.controlContainer);
+    }
+  }
+  @ContentChildren(
+    forwardRef(() => LgFilterButtonComponent),
+    {
+      descendants: true,
+    },
+  )
+  set options(options: QueryList<LgFilterButtonComponent>) {
+    if (this.variant === 'select-one') {
+      options.toArray().forEach((option: LgFilterButtonComponent) => {
+        option.control = this.control;
+      });
+    }
+    this._options = options;
+  }
+  get options(): QueryList<LgFilterButtonComponent> {
+    return this._options;
+  }
+  @ContentChild(LgValidationComponent, { static: false })
+  set errorElement(element: LgValidationComponent) {
+    this.ariaDescribedBy = this.domService.toggleIdInStringProperty(
+      this.ariaDescribedBy,
+      this._validationElement,
+      element,
+    );
+    this._validationElement = element;
+  }
+  @Input()
+  get value() {
+    return this._value;
+  }
+  set value(value) {
+    this._value = value;
+    this.onChange(value);
+    if (this.options) {
+      const selectedFilter = this.options.find(option => option.value === value);
+      if (selectedFilter && !selectedFilter.checked) {
+        selectedFilter.checked = true;
+      }
+    }
+  }
+
+  @Input()
+  get name(): string {
+    return this._name;
+  }
+  set name(value: string) {
+    this._name = value;
+    this._updateFilterButtonNames();
+  }
+
+  constructor(
+    @Self() @Optional() private control: NgControl,
+    private errorState: LgErrorStateMatcher,
+    @Optional()
+    @Host()
+    @SkipSelf()
+    private controlContainer: FormGroupDirective,
+    private domService: LgDomService,
+  ) {
+    if (this.control != null) {
+      this.control.valueAccessor = this;
+    }
+  }
+  private _name = `lg-filter-group-${nextUniqueId++}`;
+
+  @Input() id = `lg-filter-group-id-${nextUniqueId++}`;
+  @Input() disabled = false;
+  @Input() valueArrayName: string;
+  @Input() ariaDescribedBy: string;
+  @Input() variant: FilterVariant = 'select-one';
+
+  @HostBinding('class.lg-filter-group') class = true;
+
+  _options: QueryList<LgFilterButtonComponent>;
+
+  _validationElement: LgValidationComponent;
+
+  _value: string = null;
+
+  @HostListener('change', ['$event'])
+  onCheckChange(event) {
+    if (this.variant === 'select-multiple' && this.valueArrayName) {
+      const formArray: FormArray = this.controlContainer.control.get(
+        this.valueArrayName,
+      ) as FormArray;
+
+      if (event.target.checked) {
+        formArray.push(new FormControl(event.target.value));
+      } else {
+        let i = 0;
+        formArray.controls.forEach((ctrl: FormControl) => {
```

Comment:
Do you need to assign `i`? or can you do the below?
```suggestion
        formArray.controls.forEach((ctrl: FormControl, i) => {
```

Reply:
no need to assign it, worked like a charm.